src/components/EditTrackerForm/EditTrackerForm.jsx

```diff
@@ -0,0 +1,95 @@
+import React, { Component } from 'react';
+import Spin from 'antd/lib/spin';
+import Icon from 'antd/lib/icon';
+import Typography from 'antd/lib/typography';
+import { PropTypes } from 'prop-types';
+
+import * as api from '../../modules/api';
+
+import ColourOptions from '../ColourOptions/ColourOptions';
+import StatusOptions from '../StatusOptions/StatusOptions';
+
+import './EditTrackerForm.less';
+
+const { Paragraph } = Typography;
+
+class EditTrackerForm extends Component {
+	onStateChange = (index, value, type) => {
+		this.props.updateState(index, value, type);
+
+		if (type !== 'colour') {
+			const updatePromise = api.updateTracker(
+				this.props.deerStates[index]['id'],
+				{
+					id: this.props.deerStates[index]['id'],
+					animal_id:
+						type === 'name' ? value : this.props.deerStates[index]['name'],
+					status:
+						type === 'status' ? value : this.props.deerStates[index]['status'],
+					max_error_radius: this.props.deerStates[index]['max_error_radius'],
+					location_method: this.props.deerStates[index]['location_method'],
+					tracks: this.props.deerStates[index]['tracks'],
+				}
+			);
+			Promise.all([updatePromise]).then(values => {
+				return this.props.finishedLoading(index, type);
+			});
```

Comment:
With only one Promise you can do the `.then` on the actual function call. 
So: `api.updateTracker(...).then(value => {})`